docs/programming_guide/controllers/model_controller.rst

```diff
@@ -0,0 +1,223 @@
+
+.. note::
+
+    The :ref:`fl_model` object can be any type of data depending on the specific task.
+    For example, in the "train" and "validate" tasks we send the model parameters along with the task so the target clients can train and validate the model.
+    However in many other tasks that do not involve sending the model (e.g. "submit_model"), the :ref:`fl_model` can contain any type of data (e.g. metadata, metrics etc.) or may be not be needed at all.
+
+
+send_model_and_wait
+-------------------
+:func:`send_model_and_wait<nvflare.app_common.workflows.model_controller.ModelController.send_model_and_wait>` is the core communication function which enables users to send tasks to targets, and wait for responses.
+
+The ``data`` is an :ref:`fl_model` object, and the ``task_name`` is the task for the target executors to execute (Client API executors by default support "train", "validate", and "submit_model", however executors can be written for any arbitrary task name).
+
+``targets`` can be chosen from client names obtained with ``sample_clients()``.
+
+Returns the :ref:`fl_model` responses from the target clients once the task is completed (``min_responses`` have been received, or ``timeout`` time has passed).
+
+send_model
+----------
+:func:`send_model<nvflare.app_common.workflows.model_controller.ModelController.send_model>` is the non-blocking version of 
+:func:`send_model_and_wait<nvflare.app_common.workflows.model_controller.ModelController.send_model_and_wait>` with a user-defined callback when receiving responses.
+
+A callback with the signature ``Callable[[FLModel], None]`` can be passed in, which will be called when a response is received from each target.
+
+The task is standing until either ``min_responses`` have been received, or ``timeout`` time has passed.
+Since this call is asynchronous, the Controller :func:`get_num_standing_tasks<nvflare.apis.impl.controller.Controller.get_num_standing_tasks>` method can be used to get the number of standing tasks for synchronization purposes.
+
+
+Saving & Loading
+================
+
+persistor
+---------
+The :func:`save_model<nvflare.app_common.workflows.model_controller.ModelController.save_model>` and :func:`load_model<nvflare.app_common.workflows.model_controller.ModelController.load_model>`
+functions utilize the configured :class:`ModelPersistor<nvflare.app_common.abstract.model_persistor.ModelPersistor>` set in the ModelController ``persistor_id: str = "persistor"`` init argument.
+
+custom save & load
+------------------
+Users can also choose to instead create their own custom save and load functions rather than use a persistor.
+
+For example we can use PyTorch's save and load functions for the model parameters, and save the FLModel metadata with :mod:`FOBS<nvflare.fuel.utils.fobs>` separately to different filepaths.
+
+.. code-block:: python
+
+  import torch
+  from nvflare.fuel.utils import fobs
+
+  def save_model(self, model, filepath=""):
```

Comment:
minor thing though, indentation is different save_model vs load_model.

another thing, do you want to add something like the below to clarify these are in this class?:

```
class ModelController:
    ...
    def save_model(self, xxx